migration/orm_test.go

```diff
@@ -0,0 +1,154 @@
+package migration
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/errors"
+	"github.com/iov-one/weave/orm"
+	"github.com/iov-one/weave/store"
+	"github.com/iov-one/weave/weavetest/assert"
+)
+
+func TestSchemaVersionedBucket(t *testing.T) {
+	const thisPkgName = "testpkg"
+
+	reg := newRegister()
+
+	reg.MustRegister(1, &MyModel{}, NoModification)
+	reg.MustRegister(2, &MyModel{}, func(db weave.ReadOnlyKVStore, m Migratable) error {
+		msg := m.(*MyModel)
+		msg.Cnt += 2
+		return msg.err
+	})
+	reg.MustRegister(3, &MyModel{}, func(db weave.ReadOnlyKVStore, m Migratable) error {
+		panic("not implemented")
+	})
+
+	db := store.MemStore()
+
+	schema := NewSchemaBucket()
+	if _, err := schema.Create(db, &Schema{Metadata: &weave.Metadata{Schema: 1}, Pkg: thisPkgName, Version: 1}); err != nil {
+		t.Fatalf("cannot register schema version: %s", err)
+	}
+
+	b := &MyModelBucket{
+		Bucket: NewBucket(thisPkgName, "mymodel", orm.NewSimpleObj(nil, &MyModel{})),
+	}
+
+	// Use custom register instead of the global one to avoid pollution
+	// from the application during tests.
+	b.Bucket.migrations = reg
+
+	obj := orm.NewSimpleObj([]byte("schema_one"), &MyModel{
+		Metadata: &weave.Metadata{Schema: 1},
+		Cnt:      5,
+	})
+	assert.Nil(t, b.Save(db, obj))
+
+	if m, err := b.GetMyModel(db, "schema_one"); err != nil {
+		t.Fatalf("cannot get model one: %s", err)
+	} else if m.Metadata.Schema != 1 || m.Cnt != 5 {
+		t.Fatalf("unexpected result model: %#v", m)
+	}
+
+	// Storing a model with a schema version higher than currently active
+	// is not allowed.
+	obj = orm.NewSimpleObj([]byte("schema_two"), &MyModel{
```

Comment:
I would use obj2 here for more clarity. I was confused, thinking you saved the same object again. When you did save a new object.